### PR TITLE
Navigation.TopMenu: go to home page on banner click

### PR DESCRIPTION
**Pull request by xjuan**
https://phabricator.endlessm.com/T10722

### CODE REVIEW

File: js/app/utils.js [hunk 1]
```diff
@@ -36,19 +36,32 @@ let _ = Gettext.dgettext.bind(null, Config.GETTEXT_PACKAGE);
...
         widget.window.set_cursor(cursor);
         return Gdk.EVENT_PROPAGATE;
     });
-    widget.connect('leave-notify-event', function (widget) {
+    let leave = widget.connect('leave-notify-event', function (widget) {
         widget.window.set_cursor(null);
         return Gdk.EVENT_PROPAGATE;
     });
+
+    widget._hand_cursor_handlers = { enter_id: enter, leave_id: leave };
```
Comment (ptomato) [inline js/app/utils.js]: drive-by comment because I'm always on the lookout for opportunities to use the cool new ES6 syntax :wink: 
You can now do stuff like this:
```js
let enter_id = ...;
let leave_id = ...;
widget._hand_cursor_handlers = { enter_id, leave_id };
```

File: js/app/modules/navigation/topMenu.js [hunk 2]
```diff
@@ -42,17 +45,41 @@ const TopMenu = new Module.Class({
         'menu': {},
     },
 
+    _on_history_changed: function () {
+        let item = HistoryStore.get_default().get_current_item();
+
```
Comment (tchx84) [inline js/app/modules/navigation/topMenu.js]: nit; this line can go away

File: js/app/modules/navigation/topMenu.js [hunk 3]
```diff
@@ -42,17 +45,41 @@ const TopMenu = new Module.Class({
...
+        this._in_home_page = (item && item.page_type === 'home');
```
Comment (tchx84) [inline js/app/modules/navigation/topMenu.js]: use the Page consts instead of the string.
